### ISSUE TITLE
Serial USB adapter becomes erratic when ribbon cable exceeds 10 inches

### ISSUE THREAD

**Issue by gaasedelen** [XboxDev/serial-usb-adapter#6]
As the ribbon cable exceeds ~10 inches, the Serial USB adapter begins to behave erratically. This manifests in several different ways including:

- Xbox may outright FRAG when the serial adapter is attached
- WinDbg can hang, presumably waiting retrying a request for data while the SMSC keeps outputting invalid/garbage data
    - the TX (green) LED can be seen going nuts as retry attempts are made by the xbox to send the requested data
- The Xbox will not respond to any break's sent by WinDbg, so a debug session never connects.
    - RX (blue) LED blinks, but presumably the signals sent from the SMSC chip down the ribbon to the Xbox are getting corrupted / mangled by the time they reach the mobo
 - Past ~11 inches ribbon length I have never been able to successfully established a debug session

It seems like there's a signal integrity issue and was hoping someone who is an EE might be able to help diagnose what could be going wrong here.

I'm pretty confident that by the time the signals reach our SMSC chip after running along the ribbon, they're too muddy for it to decode correctly.

Some additional thoughts:
- DVT4's have longer than 10 inch ribbons running to their serial boards
- I tried cutting SDA/SCL out of the ribbon close to the LPC side see if that might help reduce 'noise' on the ribbon 
   - The ribbons in DVT4's also cut these two lines, which is why I decided to try it but it didn't help
- Is there anything that we missed about the design of the native DVT4 serial board that should be ported over to this?
- Are there going to be any issues with the fact that the FTDI side of our Serial USB circuit is powered by USB while the SMSC side is powered by the Xbox LPC? Is this best practice, or something that could also get refined in a new revision?
- This issue manifests at least on all REV A and REV C boards that I have on hand (and presumably REV B too)
- C4 is removed on all these boards. Anytime I have C4 on one of the boards, The Xbox outright FRAGs
   
 I chatted briefly with a non-Xbox EE friend who had the following to say:

> [LPC] is slow and edge rates are all that should matter here.
> 
> You can try adding 5-10 ohm resistors in series to the signal lines. it's counterintuitive, but basically it slows the rise time of the signals which removes a lot of the high frequency components. 
> 
> Basically: old school stuff had slower rise times, less high freq. components (sharper edge: more hi freq, see: fourier). higher frequencies couple better between lines (crosstalk) and will give you more issues with reflection and other transmission line effects. 
> So even if you have a newer chip with "slow" signals, it's driving the fuck out of those edges, so stick some resistors in series to slow him down, it helps sometimes 🙂
> 
> another note: you can interleave grounds to help

Would be more than happy to hear any other thoughts from HW / EE / CSE people

**Comment by gaasedelen** [comment]
TL;DR: I think the main issue is LFRAME getting too distorted by noise / reflections and adding terminating resistors of 100ohms in series on the signal lines definitely helps at longer ribbon lengths. We really should add 100ohm on all the signal lines, I think.

Specifically, I have 30cm ribbon cables from amazon (11.8 inches) that have never worked with my serial boards. They *never* emitted a TX (green) blink at these lengths, suggesting data from the Xbox to the superio board was never getting decoded.

Probing LFRAME on the motherboard is in pink, while yellow is the same signal by the time it reaches the header on the superio board:

![image](https://user-images.githubusercontent.com/9522648/176028506-373145e7-0bce-4e09-b105-88270f1ab999.png)

Adding 100ohm resistor in series to LFRAME certainly cleans it up at this length:

![image](https://user-images.githubusercontent.com/9522648/176029118-8384f502-1b64-412d-95e6-d3d17b0537f7.png)

This alone seemed to get the board able to start decoding messages from the Xbox and TX started blinking green. But I went ahead and added 100ohms to LAD[0:3] shortly after. This cable now works great with WinDBG at ~12 inches.

I also checked the clock (LCLK) which looks 'ok' (yellow at mobo, pink at end of ribbon here)

![image](https://user-images.githubusercontent.com/9522648/176029655-630856fb-6b3f-40fe-ad95-eb42727ed84c.png)

Also, this is LCLK with C4 on it (the 'bodge' capacitor) which seems to really mess up the clock signal. I leave it off my superio boards because it would cause my boxes to FRAG as soon as I added a Xenium to the ribbon. 

![image](https://user-images.githubusercontent.com/9522648/176029923-6f8bdf2e-45d2-4713-8eab-86b899cd9a2a.png)

So besides adding some terminating resistors to the signal lines, I think C4 should definitely be removed in the next revision.

I am still curious why the official DVT4 superio/cable does not suffer as badly from these issues. Maybe the ribbon or cables are better? Or maybe I have fake SMSC chips that are a bit dirtier. Not sure.

![image](https://user-images.githubusercontent.com/9522648/176032451-c432d822-97b8-4be0-945c-56fb0a06207e.png)

I daisy chained two ribbons I had on hand for ~24 inches / 60cm and the superio board seems stable at these lengths too. Should be more than sufficient,

**Comment by gaasedelen** [comment]
I just wanted to follow up and confirm that I integrated these changes to improve signal quality into a new board revision.

I removed C4, added terminating resistor networks/arrays for the LPC signal lines, and switched the board over to use a USB-C port as that's where everything is heading anyways. If you don't intend to use the board at the end of a ribbon, you should be able to just jump each of the connections instead of placing resistor networks.

Still in the process of vetting my changes but the board now works great at longer ribbon lengths.

![image](https://user-images.githubusercontent.com/9522648/195122779-12d62963-840b-4570-896a-749121687b17.png)
![image](https://user-images.githubusercontent.com/9522648/195128284-db707560-456b-4794-95fc-96a516a91df6.png)
![image](https://user-images.githubusercontent.com/9522648/195123350-fed865b2-2df2-4b7d-b4a5-1fb869ffdbef.png)
![image](https://user-images.githubusercontent.com/9522648/195123498-022b48c1-72c5-4286-95db-4071e20b3e30.png)
![image](https://user-images.githubusercontent.com/9522648/195124095-985b81cb-bd31-4247-90d9-42c6f50486d7.png)

I'm kind of a KiCad noob and may have muddied up the project files a bit (like project symbol/footprint/model libraries, paths) so I'd like to try and clean it up before making a PR. If anyone wants the gerbers/bom/centroid files before then I'd be happy to share them.
